### ISSUE TITLE
CI/CD for deployments and admin actions

### ISSUE THREAD

**Issue by Ramarti** [storyprotocol/protocol-contracts#26]
After #10 , in order to have a better dev ex among the team, we could create a CD step for merges to main. I'd recommend basing them in the work [I did in Forta (pioneered by Santiago Palladino).](https://github.com/forta-network/forta-contracts/blob/master/DEPLOYMENT.md).

We need to decide if rewriting it to use Foundry instead of Hardhat is really worth the time.

This will create an auditable upgrade/config process, initially for internal dev ex, later for multisig ops.

In short:
- Have config files/ deploy scripts that clearly define the new versions of the contracts and params
- Make sure the `IVersioned` numbers in revelant contracts are increased
- Make sure we deploy tested AND verified sources
- Use OZ Defender to securely batch config txs and upgrades into a single tx to minimize downtime.
